extensions/wikia/Privacy/RemoveUserDataOnWikiTask.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+use Wikia\Logger\Loggable;
+use Wikia\Tasks\Tasks\BaseTask;
+
+/**
+ * Class RemoveUserDataOnWikiTask
+ *
+ * Removes or anonimizes all PII for a specific user on the current community
+ *
+ */
+class RemoveUserDataOnWikiTask extends BaseTask {
+	use Loggable;
+
+	/**
+	 * Deletes all CheckUser records associated with the given user
+	 *
+	 * @param int $userId
+	 * @throws DBUnexpectedError
+	 * @return true if operation was successful
+	 */
+	private function removeCheckUserData( int $userId ) {
+		try {
+			$db = wfGetDB( DB_MASTER );
+			// remove check user data
+			$db->delete( 'cu_changes', ['cuc_user' => $userId], __METHOD__ );
+			$db->delete( 'cu_log', ['cul_target_id' => $userId], __METHOD__ );
+			$this->info( "Removed CheckUser data", ['user_id' => $userId] );
+			return true;
+		} catch( DBError $error ) {
+			$this->error( "Couldn't remove CheckUser data", ['exception' => $error, 'user_id' => $userId] );
+			return false;
+		}
+	}
+
+	/**
+	 * Removes the IP address from all RecentChanges records associated with the given user
+	 *
+	 * @param int $userId
+	 * @return true if operation was successful
+	 */
+	private function removeIpFromRecentChanges( int $userId ) {
+		try {
+			$db = wfGetDB( DB_MASTER );
+			$db->update( 'recentchanges', ['rc_ip_bin' => ''], ['rc_user' => $userId], __METHOD__ );
+			$this->info( "Removed IPs from recent changes", ['user_id' => $userId] );
+			return true;
+		} catch( DBError $error ) {
+			$this->error( "Couldn't remove IP from recent changes", ['exception' => $error, 'user_id' => $userId] );
+			return false;
+		}
+	}
+
+	/**
+	 * Removes abuse filter logs associated with the given user
+	 *
+	 * @param int $userId
+	 * @return true if operation was successful
+	 */
+	private function removeAbuseFilterData( int $userId ) {
+		global $wgEnableAbuseFilterExtension;
+		if( $wgEnableAbuseFilterExtension ) {
+			try {
+				$db = wfGetDB( DB_MASTER );
+				$db->update( 'abuse_filter', ['af_user_text' => ''], ['af_user' => $userId], __METHOD__ );
+				$db->update( 'abuse_filter_history', ['afh_user_text' => ''], ['afh_user' => $userId], __METHOD__ );
```

Comment:
the _user_text columns are always empty

Reply:
They're not, but we can remove them separately.